ugs-core/src/com/willwinder/universalgcodesender/TinyGUtils.java

```diff
@@ -206,4 +230,150 @@ private static String getStateAsString(int state) {
         ControllerState controllerState = getState(state);
         return controllerState.name();
     }
+
+    /**
+     * Generates a command for resetting the coordinates for the current coordinate system to zero.
+     *
+     * @param controllerStatus the current controller status
+     * @param gcodeState       the current gcode state
+     * @return a string with the command to reset the coordinate system to zero
+     */
+    public static String generateResetCoordinatesToZeroCommand(ControllerStatus controllerStatus, GcodeState gcodeState) {
+        int offsetCode = convertOffsetGcodeToCode(gcodeState.offset);
+        Position machineCoord = controllerStatus.getMachineCoord();
+        return "G10 L2 P" + offsetCode +
+                " X" + Utils.formatter.format(machineCoord.get(Axis.X)) +
+                " Y" + Utils.formatter.format(machineCoord.get(Axis.Y)) +
+                " Z" + Utils.formatter.format(machineCoord.get(Axis.Z));
+    }
+
+    /**
+     * Generates a command for setting the axis to a position in the current coordinate system
+     *
+     * @param controllerStatus the current controller status
+     * @param gcodeState       the current gcode state
+     * @param axis             the axis to set
+     * @param position         the position to set
+     * @return a command for setting the position
+     */
+    public static String generateSetWorkPositionCommand(ControllerStatus controllerStatus, GcodeState gcodeState, Axis axis, double position) {
+        int offsetCode = convertOffsetGcodeToCode(gcodeState.offset);
+        Position machineCoord = controllerStatus.getMachineCoord();
+        double coordinate = -(position - machineCoord.get(axis));
+        return "G10 L2 P" + offsetCode + " " +
+                axis.name() + Utils.formatter.format(coordinate);
+    }
+
+    private static int convertOffsetGcodeToCode(Code offsetGcode) {
```

Comment:
This could go right into GcodeUtils.java, maybe even some of the other WCS helpers you have here as well. There is also WorkCoordinateSystem.java, so something like this might technically work:
```
 WorkCoordinateSystem.valueOf(offsetGcode.toString()).getPValue();
```

Reply:
I'd probably lean towards keeping your static method and maybe getting rid of `WorkCoordinateSystem`